www/js/lib/uiUtil.js

```diff
@@ -1003,6 +1011,428 @@ function warnAndOpenExternalLinkInNewTab (event, clickedAnchor, archive) {
+            // (appears to be a timing issue related to the reservation of space given that the div is inserted dynamically).
+            div.innerHTML = `<div style="position: relative; overflow: hidden; height: ${div.style.height};">
+                <div style="background: ${backgroundColour} !important; opacity: 70%; position: absolute; top: 0; right: 0; display: flex; align-items: center; padding: 0; z-index: 1;">
+                    <img id="popbreakouticon" src="${breakoutIconFile}" />
+                    <span id="popcloseicon">X</span>
+                </div>
+                <div style="padding-top: 3px">${html}</div>
+            </div>`;
+            // Now it is populated, we can attach the arrow to the div
+            div.appendChild(span);
+            // Programme the icons
+            addEventListenersToPopoverIcons(link, div, currentDocument);
+            setTimeout(function () {
+                div.popoverisloading = false;
+            }, 900);
+            // Finally, return the div DOM element
+            return div;
+        }).catch(function (err) {
+            console.warn(err);
+            // Remove the div
+            div.style.opacity = '0';
+            div.parentElement.removeChild(div);
+            link.dataset.touchevoked = false;
+            link.popoverisloading = false;
+        });
+    }, 600);
+}
+
+/**
+ * Create a new empty Kiwix popover container and attach it to the current document appropriately sized and positioned
+ * in relation to the given anchor and available screen width and height. Also returns the arrow span element which can be
+ * attached to the div after the div is populated with content.
+ * @param {Window} win The window of the article DOM
+ * @param {Element} anchor The anchor element that is being actioned
+ * @param {Event} event The event which has fired this popover action
+ * @returns {Object} An object containing the popover div and the arrow span elements
+ */
+function createNewKiwixPopoverCointainer (win, anchor, event) {
+    const div = document.createElement('div');
+    const linkHref = anchor.getAttribute('href');
+    const currentDocument = win.document;
+    div.popoverisloading = true;
+    const screenWidth = win.innerWidth - 40;
+    const screenHeight = document.documentElement.clientHeight;
+    let margin = 40;
+    let divWidth = 512;
+    if (screenWidth <= divWidth) {
+        divWidth = screenWidth;
+        margin = 10;
+    }
+    // Check if we have restricted screen height
+    const divHeight = screenHeight < 512 ? 160 : 256;
+    div.style.width = divWidth + 'px';
+    div.style.height = divHeight + 'px';
+    div.style.display = 'flex';
+    div.style.justifyContent = 'center';
+    div.style.alignItems = 'center';
+    div.className = 'kiwixtooltip';
+    div.innerHTML = '<p>Loading ...</p>';
+    div.dataset.href = linkHref;
```

Comment:
Explain why we have to attach the div to the article's document in order to calculate accurate positioning.